AdminServer/appscale/admin/__init__.py

```diff
@@ -388,6 +417,66 @@ def post(self, project_id, service_id):
     self.write(json_encode(operation.rest_repr()))
 
 
+class VersionHandler(BaseHandler):
+  """ Manages particular service versions. """
+
+  def initialize(self, acc):
+    """ Defines an AppControllerClient.
+
+    Args:
+      acc: An AppControllerClient.
+    """
+    self.acc = acc
+
+  def delete(self, project_id, service_id, version_id):
+    """ Deletes a version.
+
+    Args:
+      project_id: A string specifying a project ID.
+      service_id: A string specifying a service ID.
+      version_id: A string specifying a version ID.
+    """
+    self.authenticate()
+
+    if service_id != constants.DEFAULT_SERVICE:
+      raise CustomHTTPError(HTTPCodes.BAD_REQUEST, message='Invalid service')
+
+    if version_id != constants.DEFAULT_VERSION:
+      raise CustomHTTPError(HTTPCodes.BAD_REQUEST, message='Invalid version')
+
+    try:
+      app_info_map = self.acc.get_app_info_map()
+    except AppControllerException:
+      raise CustomHTTPError(HTTPCodes.INTERNAL_ERROR,
+                            message='Unable to fetch version info')
+
+    try:
+      project_info = app_info_map[project_id]
+    except KeyError:
+      raise CustomHTTPError(HTTPCodes.NOT_FOUND,
+                            message='Specified version does not exist')
```

Comment:
No, app_info_map doesn't have version information, yet. I simplified it a bit since I don't think there is a need to differentiate between the two KeyErrors.